zerver/tests/test_saved_replies.py

```diff
@@ -0,0 +1,88 @@
+from zerver.lib.saved_replies import do_create_saved_reply, do_get_saved_replies
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.models import SavedReply, UserProfile
+
+
+class SavedReplyTests(ZulipTestCase):
+    def get_user(self) -> UserProfile:
+        user = self.example_user("hamlet")
+
+        SavedReply.objects.filter(user_profile=user).delete()
+        return user
+
+    def create_example_saved_reply(self) -> int:
+        user = self.example_user("hamlet")
+        saved_reply = do_create_saved_reply(
+            "Welcome message", "**Welcome** to the organization.", user
+        )
+        return saved_reply.id
+
+    def test_create_saved_reply(self) -> None:
+        """Tests creation of saved replies."""
+
+        user = self.get_user()
+        self.login_user(user)
+
+        self.assert_length(do_get_saved_replies(user), 0)
+
+        result = self.client_post(
+            "/json/saved_replies",
+            {"title": "Welcome message", "content": "**Welcome** to the organization."},
+        )
+        response_dict = self.assert_json_success(result)
+        saved_replies = do_get_saved_replies(user)
+        self.assert_length(saved_replies, 1)
+        self.assertEqual(response_dict["saved_reply_id"], saved_replies[0]["id"])
+
+        # Tests if the title is truncated when the length exceeds `MAX_TITLE_LENGTH`.
+        title = "This is very very long saved reply title."
+        result = self.client_post(
+            "/json/saved_replies",
+            {"title": title, "content": "Hello"},
+        )
+        response_dict = self.assert_json_success(result)
+        saved_title = SavedReply.objects.get(id=response_dict["saved_reply_id"]).title
+        self.assertEqual(saved_title, title[: SavedReply.MAX_TITLE_LENGTH])
+
+    def test_delete_saved_reply(self) -> None:
+        """Tests deletion of saved replies."""
+
+        user = self.get_user()
+        self.login_user(user)
+        saved_reply_id = self.create_example_saved_reply()
+        self.assert_length(do_get_saved_replies(user), 1)
+
+        result = self.client_delete(
+            f"/json/saved_replies/{saved_reply_id}",
+        )
+        self.assert_json_success(result)
+        self.assert_length(do_get_saved_replies(user), 0)
+
+        # Tests if error is thrown when the provided ID does not exist.
+        result = self.client_delete(
+            "/json/saved_replies/10",
+        )
+        self.assert_json_error(result, "Saved reply does not exist.")
+
+    def test_get_saved_replies(self) -> None:
+        """Tests fetching of saved replies."""
+
+        user = self.get_user()
+        self.login_user(user)
+        self.create_example_saved_reply()
+
+        result = self.client_get(
+            "/json/saved_replies",
+        )
+        saved_reply = SavedReply.objects.filter(user_profile=user).order_by("id")[0]
```

Comment:
I guess you need to be using `do_get_saved_replies` rather than filtering using the object to test `get_saved_replies`. And also, make sure to have 2 saved replies before  testing this.